components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.cpp

```diff
@@ -908,5 +910,90 @@ namespace scream {
             return band_index.hostRead();
         }
 
+        void compute_aerocom_cloudtop(
+            int ncol, int nlay, const real2d &tmid, const real2d &pmid,
+            const real2d &p_del, const real2d &z_del, const real2d &qc,
+            const real2d &qi, const real2d &cldfrac_tot, const real2d &nc,
+            real1d &T_mid_at_cldtop, real1d &p_mid_at_cldtop,
+            real1d &cldfrac_ice_at_cldtop, real1d &cldfrac_liq_at_cldtop,
+            real1d &cldfrac_tot_at_cldtop, real1d &cdnc_at_cldtop) {
+          /* The goal of this routine is to calculate properties at cloud top
+           * based on the AeroCOM recommendation. See reference for routine
+           * get_subcolumn_mask above, where equation 14 is used for the
+           * maximum-random overlap assumption for subcolumn generation. We use
+           * equation 13, the column counterpart.
+           */
+          // Set outputs to zero
+          memset(T_mid_at_cldtop, 0.0);
+          memset(p_mid_at_cldtop, 0.0);
+          memset(cldfrac_ice_at_cldtop, 0.0);
+          memset(cldfrac_liq_at_cldtop, 0.0);
+          memset(cldfrac_tot_at_cldtop, 0.0);
+          memset(cdnc_at_cldtop, 0.0);
+          // Initialize the 1D "clear fraction" as 1 (totally clear)
+          auto aerocom_clr = real1d("aerocom_clr", ncol);
+          memset(aerocom_clr, 1.0);
+          // Get gravity acceleration constant from constants
+          using physconst = scream::physics::Constants<Real>;
+          // Set non-zero q threshold
+          constexpr real q_threshold = 0.0;  // BAD_CONSTANT!
```

Comment:
Note that, even with the best intentions, things tend to left in the code longer than we think they will...rather than "BAD_CONSTANT", should we instead drop a note like "TODO: this should be tuned"? Just an idea.

Reply:
The BAD_CONSTANT terminology likely comes from EAGLES, where we have been stamping all over MAM whenever we see one (which is often!).